### PR TITLE
Arduino branch

### PR DESCRIPTION
**Pull request by jchung55**
Please review arduino_communication and test_arduino_communication

---
This change is [Reviewable](https://reviewable.io/reviews/jhu-asco/esc_closed_loop_force_control/2)

### CODE REVIEW

File: phidget_bridge.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+from Phidget22.Devices.VoltageRatioInput import VoltageRatioInput
```
Comment (garimellagowtham) [inline phidget_bridge.py]: remove this file from this branch

**Review by garimellagowtham — CHANGES_REQUESTED** [review]

File: test_phidget_bridge.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import time
```
Comment (garimellagowtham) [inline test_phidget_bridge.py]: Remove this file from this branch

File: test_arduino_commmunication.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from arduino_communication import ArduinoCommunication
+import numpy as np
+import time
+
+percentage_array = np.linspace(6,55, 5)
```
Comment (garimellagowtham) [inline test_arduino_commmunication.py]: Add space after comma

File: arduino_communication.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        self.voltage_scale = (5.0/1023.0)*10.1
+        self.port = 'com6'
+        self.baud_rate = 9600
+        self.connectingWithArduino()
+    
+    def connectingWithArduino(self, timeout = 2):
+        self.arduino = serial.Serial(self.port, self.baud_rate)
+        time.sleep(timeout)
+    
+    def sendMotorCommandtoArduino(self, percentage = 0):
```
Comment (garimellagowtham) [inline arduino_communication.py]: Rename: `sendMotorCommandToArudino`

File: arduino_communication.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import serial
+import time
+import struct
+
+class ArduinoCommunication:
+
```
Comment (garimellagowtham) [inline arduino_communication.py]: Add comments to class and functions

File: arduino_communication.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import serial
+import time
+import struct
+
+class ArduinoCommunication:
+    
+    def __init__(self):
+        self.voltage_scale = (5.0/1023.0)*10.1
+        self.port = 'com6'
```
Comment (garimellagowtham) [inline arduino_communication.py]: Take the port and baud rate as arguments to the constructor as in `__init__(self, port='com6', baud_rate=9600)`
You do not have to store the baud rate and com port one you merge the `connectingWithArduino` function with the constructor

File: arduino_communication.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        self.baud_rate = 9600
+        self.connectingWithArduino()
+    
+    def connectingWithArduino(self, timeout = 2):
```
Comment (garimellagowtham) [inline arduino_communication.py]: We do not need this function Merge this with constructor

**Review by garimellagowtham — CHANGES_REQUESTED** [review]

File: test_arduino_commmunication.py [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+arduino_communication = ArduinoCommunication()
+for i in range(0, len(percentage_array), 1):
+    percent = percentage_array[i]
+    arduino_communication.sendMotorCommandtoArduino(percent)
+    time.sleep(2)
+    print(arduino_communication.getVoltageFromArduino())
+arduino_communication.sendMotorCommandtoArduino(0)
+arduino_communication.closeArduino()
```
Comment (garimellagowtham) [inline test_arduino_commmunication.py]: Add a new line at end of file

**Comment by msheckells** [comment]
Reviewed 2 of 4 files at r1.
Review status: 2 of 4 files reviewed at latest revision, 9 unresolved discussions.

---

*[arduino_communication.py, line 8 at r1](https://reviewable.io:443/reviews/jhu-asco/esc_closed_loop_force_control/2#-KomjhXF_fd5-iq2Rd4b:-KomjhXGQQBwYM70z33x:btup4ky) ([raw file](https://github.com/jhu-asco/esc_closed_loop_force_control/blob/472d0f77d31ec226f1073c515e8eeeb0ef9906d9/arduino_communication.py#L8)):*
> ```Python
>     
>     def __init__(self):
>         self.voltage_scale = (5.0/1023.0)*10.1
> ```

Would be good to store special numbers in their own variables with descriptive names, e.g. `self.adc_conversion_factor = 5.0/1023.0`.

At the very least add a comment about what the numbers above correspond to.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/jhu-asco/esc_closed_loop_force_control/2)*

**Review by garimellagowtham — APPROVED** [review]

**Pull request merged**